src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftPermanentStorage.java

```
@@ -208,26 +227,106 @@ public boolean delete(SapsImage task) {
-        return null;
+    public List<AccessLink> generateAccessLinks(String taskId)
+        throws TaskNotFoundException, IOException {
+        List<String> files = this.listFiles(taskId);
+        List<AccessLink> filesLinks = this.generateLinks(files);
+        return filesLinks;
+    }
+
+    private List<String> listFiles(String taskId) throws IOException, TaskNotFoundException {
+        List<String> files = new ArrayList<>();
+
+        String[] dirs = {INPUTDOWNLOADING_DIR, PREPROCESSING_DIR, PROCESSING_DIR};
+
+        if(!this.swiftAPIClient.existsTask(this.containerName, this.tasksDirName, taskId)) {
+            throw new TaskNotFoundException("Task [" + taskId + "] was not found in directory [" + this.tasksDirName + "] of container [" + this.containerName + "]");
+        }
+
+        for (String dir : dirs) {
+            String dirPath = String.format(SWIFT_TASK_STAGE_DIR_PATTERN, this.tasksDirName, taskId, dir);
+            try {
+                files.addAll(this.swiftAPIClient.listFiles(this.containerName, dirPath));
+            } catch (IOException e) {
+                LOGGER.error("Error while list files of path [" + dir + "] from Object Storage", e);
+            }
+        }
+        return files;
+    }
+
+    private List<AccessLink> generateLinks(List<String> filesPaths) {
+        List<AccessLink> filesLinks = new ArrayList<>();
+        for(String filePath : filesPaths) {
+            String link;
+            try {
+                link = generateTempURL(filePath);
+                Path p = Paths.get(filePath);
+                String name = p.getParent().getFileName().toString() + "/" + p.getFileName().toString();
+                filesLinks.add(new AccessLink(name, link));
+            } catch (NoSuchAlgorithmException | InvalidKeyException e) {
+                LOGGER.error("Error while generate temp url to File Path [" + filePath + "]", e);
```

should we fail silently in such conditions?

---

Fixed.